### ISSUE TITLE
Compiler error in ScalableProblems example using MinGW g++ 4.3.0

### ISSUE THREAD

**Issue by svigerske** [coin-or/Ipopt#84]
Issue created by migration from Trac.

Original creator: guest

Original creation time: 2008-10-07 19:25:14

Assignee: ipopt-team

Version: 3.5

When trying to build the executable in examples/ScalableProblems, I got the following error message:
```
$ make

g++ -O3 -fomit-frame-pointer -pipe -DNDEBUG -pedantic-errors -Wimplicit
-Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith
-Wwrite-strings -Wconversion -Wno-unknown-pragmas   -I`echo
/c/GNU/Ipopt-3.5.4/include/coin` -c -o solve_problem.o `test -f
'solve_problem.cpp' || echo './'`solve_problem.cpp

solve_problem.cpp: In function 'int main(int, char**)':

solve_problem.cpp:140: error: 'strcmp' was not declared in this scope

make: *** [solve_problem.o] Error 1
```
----
The function 'strcmp' is really a member of cstring/string.h, which is
not explicitly included in 'solve_problem.cpp'. I think that most
compilers implicitly include cstring/string.h when cstdlib/stdlib.h is
included, and therefore this error normally does not show up. However,
to be on the safe side with all compilers, cstring/string.h needs to be
explicitly included.

To successfully compile, I simply added the following code to 'solve_problem.cpp':
```
#ifdef HAVE_CSTRING
# include <cstring>
#else
# ifdef HAVE_STRING_H
#  include <string.h>
# else
#  error "don't have header file for string"
# endif
#endif
```

**Comment by svigerske** [comment]
Comment by guest created at 2008-10-07 19:27:19

Above report was entered by Anders Gustafsson, anders dot 9ustafsson at gmail dot com.

**Comment by svigerske** [comment]
Comment by guest created at 2008-10-07 19:48:49

According to this link [http://gcc.gnu.org/gcc-4.3/porting_to.html], there has been a massive cleanup of header dependencies in GCC 4.3. As far as I understand, it should be OK to #include <cstring> even when using other compilers.

/ Anders Gustafsson

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2008-10-07 23:04:09

Hi Anders,

I also included this bugfix in trunk and stable/3.5.

Greetings,

Andreas

**Comment by svigerske** [comment]
Comment by @andrea5w created at 2008-10-07 23:04:09

Resolution: fixed
